src/amqpvalue.c

```diff
@@ -6511,8 +6561,19 @@ static int internal_decoder_decode_bytes(INTERNAL_DECODER_DATA* internal_decoder
                                 uint32_t i;
 
                                 internal_decoder_data->decode_to_value->value.map_value.pair_count /= 2;
+                                size_t malloc_size = safe_multiply_size_t(sizeof(AMQP_MAP_KEY_VALUE_PAIR), ((size_t)internal_decoder_data->decode_to_value->value.map_value.pair_count);
```

Comment:
Nit - not sure why we divide the pair_count by two on first line of this block and then `malloc_size = safe_multiply_size_t(malloc_size, 2);` a few lines later?  This is pre-existing code so maybe outside scope of this change though.